apps/token-e2e/src/integration/view/wallet-vega.cy.js

```diff
@@ -0,0 +1,285 @@
+const currencyValue = '[data-testid="currency-value"]';
+const vegaUnstaked = '[data-testid="vega-wallet-balance-unstaked"] .text-right';
+const governanceBtn = '[href="/governance"]';
+const stakingBtn = '[href="/staking"]';
+const manageLink = '[data-testid="manage-vega-wallet"]';
+const dialogWalletName = `[data-testid="key-${Cypress.env(
+  'vegaWalletPublicKey'
+)}"] h2`;
+const dialogVegaKey = '[data-testid="vega-public-key-full"]';
+const dialogDisconnectBtn = '[data-testid="disconnect"]';
+const copyPublicKeyBtn = '[data-testid="copy-vega-public-key"]';
+
+context('Vega Wallet - verify elements on widget', function () {
+  before('visit token home page', function () {
+    cy.visit('/');
+  });
+
+  describe('with wallets disconnected', function () {
+    before('wait for widget to load', function () {
+      cy.get(walletContainer, { timeout: 10000 }).should('be.visible');
+    });
+
+    it('should have VEGA WALLET header visible', function () {
+      cy.get(walletContainer).within(() => {
+        cy.get(walletHeader)
+          .should('be.visible')
+          .and('have.text', 'Vega Wallet');
+      });
+    });
+
+    it('should have Connect Vega button visible', function () {
+      cy.get(walletContainer).within(() => {
+        cy.get(connectButton)
+          .should('be.visible')
+          .and('have.text', 'Connect Vega wallet to use associated $VEGA');
+      });
+    });
+
+    it('should have Get a Vega wallet link visible', function () {
+      cy.get(walletContainer).within(() => {
+        cy.get(getVegaLink)
+          .should('be.visible')
+          .and('have.text', 'Get a Vega wallet')
+          .and('have.attr', 'href', 'https://vega.xyz/wallet');
+      });
+    });
+  });
+
+  describe('when connect button clicked', function () {
+    before('click connect vega wallet button', function () {
+      cy.get(walletContainer).within(() => {
+        cy.get(connectButton).click();
+      });
+    });
+
+    it('should have Connect Vega header visible', function () {
+      cy.get(dialog).within(() => {
+        cy.get(dialogHeader)
+          .should('be.visible')
+          .and('have.text', 'Connect to your Vega Wallet');
+      });
+    });
+
+    it('should have REST connector visible on list', function () {
+      cy.get(connectorsList).within(() => {
+        cy.get('button').should('be.visible').and('have.text', 'rest provider');
+      });
+    });
+
+    it('should have close button visible', function () {
+      cy.get(dialog).within(() => {
+        cy.get(dialogCloseBtn).should('be.visible');
+      });
+    });
+  });
+
+  describe('when rest connector form opened', function () {
+    before('click rest provider link', function () {
+      cy.get(connectorsList).within(() => {
+        cy.get('button').click();
+      });
+    });
+
+    it('should have url field visible', function () {
+      cy.get(restConnectorForm).within(() => {
+        cy.get(restUrl).should('be.visible');
```

Comment:
Here I think we can check to make sure the url is `http://localhost:1789/api/v1`. This should be the same for all environments